crates/pallet-rewards/src/lib.rs

```diff
@@ -97,40 +212,222 @@ mod pallet {
 
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-        fn on_initialize(block_number: BlockNumberFor<T>) -> Weight {
-            Self::do_initialize(block_number);
-            T::WeightInfo::on_initialize()
-        }
-
         fn on_finalize(now: BlockNumberFor<T>) {
             Self::do_finalize(now);
         }
     }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Update dynamic issuance parameters
+        #[pallet::call_index(0)]
+        #[pallet::weight(T::WeightInfo::update_issuance_params(proposer_subsidy_points.len() as u32, voter_subsidy_points.len() as u32))]
+        pub fn update_issuance_params(
+            origin: OriginFor<T>,
+            proposer_subsidy_points: BoundedVec<
+                RewardPoint<BlockNumberFor<T>, BalanceOf<T>>,
+                T::MaxRewardPoints,
+            >,
+            voter_subsidy_points: BoundedVec<
+                RewardPoint<BlockNumberFor<T>, BalanceOf<T>>,
+                T::MaxRewardPoints,
+            >,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            ProposerSubsidyPoints::<T>::put(proposer_subsidy_points);
+            VoterSubsidyPoints::<T>::put(voter_subsidy_points);
```

Comment:
> Although this call requires sudo, I think some check on the subsidy points is necessary to ensure it is sorted.

I was actually trying to avoid that here. If it is not sorted, it will simply result in no issuance due to start/end being non-increasing. This is supposed to be called VERY rarely with carefully selected values, thus probability of getting something that is not sorted is basically zero for practical purposes.

@dariolina any concerns from you here?

Reply:
Because it's sudo I think we can consider it trusted. @nazar-pc what was the reason you try to avoid the checks here?
>point.block += reward_enabled_block_number?
Actually we want to offset this by the block in which the update rolled out. This was so in a previous of this PR,  is still done?

Reply:
> what was the reason you try to avoid the checks here?

Simpler implementation (including testing/benchmarking).

> Actually we want to offset this by the block in which the update rolled out. This was so in a previous of this PR, is still done?

This is taken care of by the migration in `subspace-runtime` if I understand you correctly.

Reply:
Right, I see it in subspace-runtime now, I remembered seeing it was already handled.
@NingLin-P if you mean it has to be offset by the block when the rewards were initially enabled before this upgrade then no, that becomes irrelevant as soon as the upgrade comes through and the block of the upgrade becomes the starting point of dynamic issuance.